remotes/docker/fetcher.go

```diff
@@ -150,6 +151,28 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 	})
 }
 
+func httpError(req *request, resp *http.Response) error {
+	var registryErr Errors
+
+	buf := &bytes.Buffer{}
+	_, err := io.Copy(buf, resp.Body)
```

Comment:
Can we use `io.LimitReader` ?

Reply:
Sure. How about liming the body around 1024 bytes? It is pretty arbitrary though.